### PR TITLE
Merges WCAndroid changes to the login library from base repository

### PR DESCRIPTION
**Pull request by oguzkocer**
wordpress-mobile/WordPress-Login-Flow-Android#57 merges changes from both WPAndroid and WCAndroid for login library. This PR brings in those changes so both clients and the base repo is in the same state.

I'll add line comments for where each change has been made in WCAndroid. Note that, this does NOT mean that every WCAndroid change is the right one for WPAndroid. Part of this PR review/testing process is to figure out if there has been any changes to the library that's not valid for either client. Please make sure to pay extra attention to this review.

**To test:**
* This was a very error prone merge, so I think we should test the whole login flow

## Regression Notes
1. Potential unintended areas of impact
* Breaking login flow

2. What I did to test those areas of impact (or what existing automated tests I relied on)
* We'll run extensive manual tests for login flow
* The existing login automated tests would also be helpful in ensuring the flow works correctly

3. What automated tests I added (or what prevented me from doing so)
N/A

**PR submission checklist:**

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: libs/login/WordPressLoginFlow/build.gradle [hunk 1]
```diff
@@ -21,10 +21,10 @@ apply plugin: 'kotlin-kapt'
 apply plugin: 'kotlin-android-extensions'
 
 repositories {
+    mavenCentral()
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/build.gradle:24]: The build file changes have been made as part of several WCAndroid PRs. Most of them are version upgrades, so as long as we test the login flow, should be good to go. The `mavenCentral` addition and the `bintray` maven url removal are related to Bintray shutdown.

File: libs/login/WordPressLoginFlow/build.gradle [hunk 2]
```diff
@@ -47,14 +47,14 @@ dependencies {
...
+    implementation 'androidx.media:media:1.2.1'
     implementation 'androidx.legacy:legacy-support-v13:1.0.0'
     implementation 'androidx.gridlayout:gridlayout:1.0.0'
+
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
     implementation "com.google.android.material:material:1.2.1"
-    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
 
     implementation "androidx.core:core-ktx:$kotlin_ktx_version"
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/build.gradle:57]: https://github.com/woocommerce/woocommerce-android/pull/3751/files#r602173123

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseFormFragment.java [hunk 3]
```diff
@@ -239,7 +239,9 @@ protected void startProgress() {
     }
 
     protected void startProgress(boolean cancellable) {
-        mBottomButton.setEnabled(false);
+        if (mBottomButton != null) {
+            mBottomButton.setEnabled(false);
+        }
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseFormFragment.java:244]: https://github.com/woocommerce/woocommerce-android/pull/3803

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginEmailFragment.java [hunk 4]
```diff
@@ -167,8 +167,9 @@ protected void setupLabel(@NonNull TextView label) {
                 break;
             case WOO_LOGIN_MODE:
                 if (mOptionalSiteCredsLayout) {
+                    String siteAddressClean = mLoginSiteUrl.replaceFirst("^(http[s]?://)", "");
                     label.setText(Html.fromHtml(
-                            getString(R.string.enter_email_for_site, mLoginSiteUrl)));
+                            getString(R.string.enter_email_for_site, siteAddressClean)));
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginEmailFragment.java:172]: https://github.com/woocommerce/woocommerce-android/pull/3363

File: libs/login/WordPressLoginFlow/src/main/res/layout/toolbar_login.xml [hunk 5]
```diff
@@ -17,9 +17,9 @@
             android:layout_height="24dp"
             android:layout_gravity="center"
             android:contentDescription="@null"
-            android:tint="?attr/colorControlNormal"
             android:visibility="gone"
-            app:srcCompat="@drawable/login_toolbar_icon" />
+            app:srcCompat="@drawable/login_toolbar_icon"
+            app:tint="?attr/colorControlNormal" />
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/res/layout/toolbar_login.xml:22]: This is just a change from `android:tint` to `app:tint` I've manually made to fix the lint error in the library.

File: libs/login/gradlew.bat [hunk 6]
```diff
@@ -29,9 +29,6 @@ if "%DIRNAME%" == "" set DIRNAME=.
 set APP_BASE_NAME=%~n0
```
Comment (oguzkocer) [inline libs/login/gradlew.bat:29]: The changes in this file have been made by running `./gradlew wrapper --gradle-version 6.1.1 --distribution-type all` and should be ignored.

**Review by zwarm — APPROVED** [review]
**Tested the following scenarios in WPAndroid (Jetpack variant)**
✅ WordPress.com account with Jetpack sites: Login with Continue with WordPress.com -> Enter valid email address + password -> Epilogue view is shown
✅ WordPress.com account with NO Jetpack sites: Login with Continue with WordPress.com -> Enter valid email address + password -> Correct error message is displayed.
✅ Login with Continue with WordPress.com -> Enter valid email address + magic link login -> Epilogue view is shown
✅ Enter Site Address: Jetpack site that's connected to WordPress.com: Login with Enter your site address -> Login with WP.com view is shown
✅ Enter Site Address: Jetpack site that's NOT connected to WordPress.com -> Enter valid email address -> Correct error message is displayed.
✅ Enter Site Address: Site that has Jetpack installed but not activated -> Enter valid email address -> Correct error message is displayed.
✅ Entered a non WP site and correct error message is displayed.
✅ Reset password works as expected.


**Tested the following scenarios in WPAndroid (WordPress variant)**
✅ Login with Continue with WordPress.com -> Enter valid email address + password
✅ Login with Continue with WordPress.com -> Enter valid email address + magic link login
✅ Login with Enter using your existing site address -> Enter valid email address + password
✅ Login with Enter using your existing site address -> Enter valid email address + magic link login
✅ Login with Enter using your existing site address -> Enter a url that does not have WordPress -> Error message is displayed.
✅ Login with Enter using your existing site address -> Enter an invalid username -> correct error message is displayed
✅ Reset password works as expected.
✅ Entered a non WP site and error message is displayed.

All works as expected. 👍

**Comment by develric** [comment]
Tested the WPAndroid login flows on a OnePlus 7T device with Android 11.

As a guideline I used the test scenarios reported by Renan here `pbMoDN-ET-p2`, adding the `reset password` scenario. 

No issues found in my tests, also changes look good to me. Thanks for wrangling this tricky task @oguzkocer 🙇 !

**Review by develric — APPROVED** [review]

**Comment by renanferrari** [comment]
@oguzkocer Thanks for taking care of this! 🙇 

I've tested most of the scenarios @develric mentioned and everything seems to be working as expected. Code-wise, most of the changes seem harmless, so I believe we're good to go 👍

**Review by renanferrari — APPROVED** [review]

**Pull request merged**